### PR TITLE
Improve BOOL

### PR DESCRIPTION
**Pull request by rylev**

### CODE REVIEW

**Review by kennykerr — APPROVED** [review]
Yes, that seems more natural - thanks!

File: src/result/bool32.rs [hunk 1]
```diff
@@ -39,18 +44,54 @@ unsafe impl Abi for BOOL {
     type Abi = Self;
 }
 
+impl From<BOOL> for bool {
+    fn from(value: BOOL) -> Self {
+        value == TRUE
```
Comment (lespea) [inline src/result/bool32.rs:49]: I'm far from a C expert, but shouldn't this be `value != FALSE` or `value > FALSE`?  I thought any non-zero result was true?  Or is this a "windows bool" thing?

Comment (kennykerr) [reply src/result/bool32.rs:49]: Well spotted - I thought I'd checked every comparison. Indeed, we should always compare against zero/false.

Comment (rylev) [reply src/result/bool32.rs:49]: This makes me think we should not derive `PartialEq` and instead normalize when doing equality checks.

Comment (kennykerr) [reply src/result/bool32.rs:49]: Fixes here: https://github.com/microsoft/windows-rs/pull/492

**Pull request merged**